### DISCUSSION TITLE
write-all! (force write all) is not working

### DISCUSSION THREAD

**Discussion by 7flash** [helix-editor/helix#11698] [Troubleshooting]
### Summary

When I type :wa! nothiing happens therefore I have to go through each individual file and do :w!

TLDR; :wa and :w! works but :wa! is not working

### Platform

macos

### Terminal Emulator

kitty

### Installation Method

brew

### Helix Version

helix 23.10

**Comment by 7flash** [comment]
#7556

**Comment by the-mikedavis** [comment]
I can't reproduce this: `:wa!` saves any files that I make changes to. Can you give a reproduction case and more details?

**Reply by 7flash** [reply]
I should have mentioned, the issue appear when there are external changes to my file, and I want to force-override them; so to reproduce, try to make changes to currently opened file in external editor, and then compare outcomes of ":w!" which will override as expected vs ":wa!" which will not unfortunately

**Reply by the-mikedavis** [reply]
It looks like we skip documents that aren't modified since their last save within Helix: https://github.com/helix-editor/helix/blob/5ce77de0dc7106c6f1460d80a3c5a51eaea3108c/helix-term/src/commands/typed.rs#L690-L692

That condition could consider the `force` parameter so that we save documents that haven't been modified as well.
